knowledge-base/update-toc-radwordsprocessing-before-docx-export.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Updating TOC Page Numberings in Word Documents Before Exporting to DOCX Format
+description: Learn how to update the Table of Contents (TOC) in a Word document using RadWordsProcessing before exporting to DOCX format.
+type: how-to
+page_title: How to TOC Page Numberings in Word Documents Before Exporting to DOCX Format
+slug: update-toc-radwordsprocessing-before-docx-export
+tags: wordsprocessing, document, processing, toc, table, contents, update, fields, docx, export
+res_type: kb
+ticketid: 1657401
+---
+
+## Environment
+
+| Product | Version |
+| --- | --- |
+| RadWordsProcessing for Document Processing | 18.0.24.305 |
+
+## Description
+
+When working with Word documents that contain a [Table of Contents (TOC)]({%slug radwordsprocessing-concepts-toc-field%}), it is often necessary to update the TOC to reflect the correct page numbers before exporting the document to DOCX format. This KB article shows what steps are needed to refresh the page numbers in a document's TOC before exporting the flow document content to DOCX format.
+
+|Before|After|
+|----|----|
+|![TOC Before](images/toc-before.png) |![TOC After](images/toc-after.png) |   
+
+## Solution
+
+To update the TOC in a [RadFlowDocument]({%slug radwordsprocessing-model-radflowdocument%}) before exporting it to DOCX format, follow these steps:
+
+1. Ensure that your document contains a [TOC field]({%slug radwordsprocessing-concepts-toc-field%}). RadWordsProcessing supports the TOC field and allows for its update before exporting the document.
+
+2. Before exporting the document to PDF, use the `UpdateFields()` method to update all fields, including the TOC, in the `RadFlowDocument`. This ensures that the TOC reflects the correct page numbers.
+
+3. To accurately update the TOC, including the correct page numbering, it's necessary to calculate the layout of the document. RadWordsProcessing provides the [NumberingFieldsProvider]({%slug radpdfprocessing-formats-and-conversion-pdf-numbering-fields-provider%}) for this purpose.
+
+```csharp
+FlowExtensibilityManager.NumberingFieldsProvider = new NumberingFieldsProvider();
+document.UpdateFields();
```

Comment:
maybe also add an example of how to update a single field (in this case the "TOC" field)